src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Capture.cs

```diff
@@ -7,15 +7,24 @@
 // search may return multiple Capture within each capturing
 // RegexGroup.
 
+using System.Buffers;
+
 namespace System.Text.RegularExpressions
 {
     /// <summary>
     /// Represents the results from a single subexpression capture. The object represents
-    /// one substring for a single successful capture.
+    /// the input slice for a single successful capture.
     /// </summary>
     public class Capture
     {
-        internal Capture(string text, int index, int length)
+        private string _value;
+
+        /// <summary>
+        /// Creates a Capture with an input string, a start index of the capture
+        /// and the length of the captured string.
+        /// </summary>
+        /// <param name="text">The input string provided by the user wrapped in a Memory.</param>
+        internal Capture(in MemoryOrPinnedSpan<char> text, int index, int length)
```

Comment:
Does `in` make a difference perf-wise?

Reply:
No idea, haven't measured... Isn't the recommendation that structs which are `readonly` and >= `sizeof(IntPtr)` always benefit from `in`?

Reply:
Nope. The struct has to be typically much larger than IntPtr to benefit from `in`. MemoryOrPinnedSpan is likely above the threshold, but you can never tell for sure until you measure.

Reply:
Interesting! I probably won't benchmark this change till the design review but definitely during the implementation phase later. thanks.